### PR TITLE
Flash size reduction for mime-type

### PR DESCRIPTION
**Pull request by s-hadinger**
Superseded #7311 

The current mime-type array is based on `Entry` structure with fixed size string that waste a lot of Flash space (~400 bytes):
```
struct Entry
{
  const char endsWith[16]; 
  const char mimeType[32];
};
```

I replaced with standard PROGMEM strings.

I also added `#define MIMETYPE_MINIMAL` to reduce the footprint to mime-types that are strictly necessary: `html`, `txt`, `gz`

Overall this saves 1kB for Tasmota.

### CODE REVIEW

File: libraries/ESP8266WebServer/src/detail/mimetable.cpp [hunk 1]
```diff
@@ -5,48 +5,93 @@
 namespace mime
 {
 
-// Table of extension->MIME strings stored in PROGMEM, needs to be global due to GCC section typing rules
-const Entry mimeTable[maxType] PROGMEM = 
+static const char kHtmlSuffix[] PROGMEM = ".html";
+static const char kHtmSuffix[] PROGMEM = ".htm";
+static const char kCssSuffix[] PROGMEM = ".css";
```
Comment (devyte) [inline libraries/ESP8266WebServer/src/detail/mimetable.cpp:10]: Missing MIMETYPE_MINIMAL conditional

File: libraries/ESP8266WebServer/src/detail/mimetable.cpp [hunk 2]
```diff
@@ -5,48 +5,93 @@
...
+static const char kSfntSuffix[] PROGMEM = ".sfnt";
+static const char kXmlSuffix[] PROGMEM = ".xml";
+static const char kPdfSuffix[] PROGMEM = ".pdf";
+static const char kZipSuffix[] PROGMEM = ".zip";
+static const char kGzSuffix[] PROGMEM = ".gz";
+static const char kAppcacheSuffix[] PROGMEM = ".appcache";
+static const char kDefaultSuffix[] PROGMEM = "";
+
+static const char kHtml[] PROGMEM = "text/html";
+static const char kCss[] PROGMEM = "text/css";
```
Comment (devyte) [inline libraries/ESP8266WebServer/src/detail/mimetable.cpp:34]: Missing MIMETYPE_MINIMAL conditional

File: libraries/ESP8266WebServer/src/detail/mimetable.cpp (hunk 1 above)
Comment (s-hadinger) [reply libraries/ESP8266WebServer/src/detail/mimetable.cpp:10]: The conditional is not strictly needed, the linker will remove the strings that are not referenced. I can add an explicit conditional.

Comment (s-hadinger) [reply libraries/ESP8266WebServer/src/detail/mimetable.cpp]: Conditional added.

**Comment by earlephilhower** [comment]
One of the VMs died during the CI process.  Just restarted, once it's good I'll hit merge.

**Pull request merged**